### PR TITLE
Add stream name in handler context

### PR DESCRIPTION
**Pull request by prolic**

### CODE REVIEW

File: src/Projection/AbstractProjection.php [hunk 1]
```diff
@@ -133,10 +131,13 @@ protected function handleStreamWithHandlers(string $streamName, Iterator $events
         foreach ($events as $event) {
             /* @var Message $event */
             $this->position->inc($streamName);
+
             if (! isset($this->handlers[$event->messageName()])) {
                 continue;
             }
+
             $handler = $this->handlers[$event->messageName()];
+            $handler = Closure::bind($handler, $this->createHandlerContext($streamName));
```
Comment (prolic) [inline src/Projection/AbstractProjection.php]: Or is this no performance problem at all?

Comment (prolic) [inline src/Projection/AbstractProjection.php]: Perhaps we add stream name as third argument to the handler, but this is something I like to avoid.

Comment (codeliner) [inline src/Projection/AbstractProjection.php]: handler context could be created outside of the `foreach` and with a `setStreamName` method but I would only do this if it has really positive influence on the performance.

Comment (codeliner) [inline src/Projection/AbstractProjection.php]: ah forget my idea. Immutability is king.

**Pull request merged**